### PR TITLE
juan/W-12651281

### PR DESCRIPTION
**Pull request by JuanStenghele-sf**
### What does this PR do?

Updates the CLI to the new promote API response.

### What issues does this PR fix or reference?

[W-12651281](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001MgWKtYAN/view)

### Functionality Before

It handled a string API response.

### Functionality After

It now handles JSON responses.

### CODE REVIEW

File: src/common/base/abstractPromote.ts [hunk 1]
```diff
@@ -207,7 +207,24 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
...
     };
-    return targetOrg.getConnection().request(req);
+    try {
+      const response: ApiPromoteResponse = await targetOrg.getConnection().request(req);
+      return response.jobId;
+    } catch (error) {
+      // The error message contains the API response
+      // We should parse that to get the error
+      const rawError = error as ApiError;
+      try {
```
Comment (JuanStenghele-sf) [inline src/common/base/abstractPromote.ts]: This second try is weird

File: src/common/types.ts [hunk 2]
```diff
@@ -43,9 +43,14 @@ export enum TestLevel {
...
 export interface ApiError extends Error {
   errorCode: string;
-  name: string;
 }
 
+export interface ApiPromoteResponse {
+  jobId: string;
+}
+
+export type ApiQuickResponse = ApiPromoteResponse;
```
Comment (EstebanRomero84) [inline src/common/types.ts]: Why do we need a specific type for quick? It is calling the same api and receiving the same response than the "regular" promote.

Comment (JuanStenghele-sf) [reply src/common/types.ts]: Ok! We can use the same type. I made it like this to differentiate between the two operations, but we can change this.

**Review by EstebanRomero84 — APPROVED** [review]
This looks fine.
Thanks Juan.

**Pull request merged**